experimental/experimental.go

```diff
@@ -2,3 +2,17 @@
 //
 // Note: All features here may be changed or deleted at any time, so use with caution!
 package experimental
+
+import (
+	"github.com/tetratelabs/wazero/api"
+)
+
+// InternalModule exposes additional module information not available through
+// api.Module.
+type InternalModule interface {
+	// GlobalsCount returns the count of all globals in the module.
+	GlobalsCount() int
+
+	// ViewGlobal provides a read-only view for a given global index. Panics if idx > GlobalsCount().
+	ViewGlobal(idx int) api.Global
+}
```

Comment:
I'd make a few changes here, tho some might be opinionated, so others may have different perspectives.

- Let's have `InternalModule` implement `api.Module`, since it's intended to be an extension of it
- Unless we've used the *Count name elsewhere in the wazero APIs, how about naming the method `NumGlobal` (e.g. `reflect.Type` uses `NumField`, `NumMethod`, etc...)
- Rename `ViewGlobal` to just `Global`

```go
type InternalModule interface {
    api.Module
    
    NumGlobal() int
    
    Global(i int) api.Global
}
```

Reply:
> Unless we've used the *Count name elsewhere in the wazero APIs, how about naming the method NumGlobal (e.g. reflect.Type uses NumField, NumMethod, etc...)

Good idea. On a quick search, I don't see any precedent of either.

Reply:
> Rename ViewGlobal to just Global

Happy to. I wrote it that way to signal that it's going to be read-only (if you know that a given global should be mutable, you're still not going to be able to edit it through this API).